core/src/main/java/io/camunda/migrator/impl/RuntimeValidator.java

```diff
@@ -87,9 +88,9 @@ public void validateProcessInstanceState(String legacyRootProcessInstanceId, Con
   public void validateC7FlowNodes(String processDefinitionId, String activityId) {
     BpmnModelInstance c7BpmnModelInstance = repositoryService.getBpmnModelInstance(processDefinitionId);
     FlowElement element = c7BpmnModelInstance.getModelElementById(activityId);
-    if ((element instanceof Activity activity)
-        && (activity.getLoopCharacteristics() instanceof MultiInstanceLoopCharacteristics)) {
-      throw new IllegalStateException(String.format(MULTI_INSTANCE_LOOP_CHARACTERISTICS_ERROR, element.getId()));
+
+    if (isMultiInstanceActivity(activityId, element)) {
+      throw new IllegalStateException(String.format(MULTI_INSTANCE_LOOP_CHARACTERISTICS_ERROR, activityId));
```

Comment:
The log statement is still not ideal:
```
Skipping process instance with legacyId [c37b0d8d-63e4-11f0-8c26-5682f3a431b7]: Found multi-instance loop characteristics for flow node with id [ServiceTask_1#multiInstanceBody] in C7 process instance.
```

The user searches for the flow node ID `ServiceTask_1#multiInstanceBody` in the BPMN model but cannot find it since the `#multiInstanceBody` is virtual and doesn't actually exist in the BPMN model. I would vote for removing the `#multiInstanceBody` from the activity ID in the log statement.

Reply:
Created PR: https://github.com/camunda/camunda-7-to-8-data-migrator/pull/180